src/MockBot.UnitTests/Extensions/ServiceCollectionExtensionTests.cs

```diff
@@ -0,0 +1,106 @@
+
+            // Act & Assert
+            _ = Assert.Throws<ArgumentNullException>(() => services.AddDmrCommunications(null));
+        }
+
+        [Fact]
+        public void AddDmrCommunicationsAddsCorrectTypes()
+        {
+            // Arrange
+            IServiceCollection services = new ServiceCollection();
+
+            var inMemorySettings = new Dictionary<string, string> {
+                {"DmrServiceSettings:CentOpsApiKey", "value"},
+                {"DmrServiceSettings:CentOpsUrl", "value"},
+            };
+
+            IConfiguration configuration = new ConfigurationBuilder()
+                .AddInMemoryCollection(inMemorySettings)
+                .Build();
+
+            // Act
+            services.AddDmrCommunications(configuration);
+
+            // Assert
+            // Ensure that there's an ICentOpsService Implementation otherwise code needs to change.
+            var interfaces = services.Select(s => s.ServiceType).ToArray();
+            _ = interfaces.Should().Contain(typeof(ICentOpsService));
+        }
+
+        [Fact]
+        public void AddApiAuthenticationThrowsForMissingConfiguration()
+        {
+            // Arrange
+            IServiceCollection services = new ServiceCollection();
+
+            // Act & Assert
+            _ = Assert.Throws<ArgumentNullException>(() => services.AddApiAuthentication(null));
+        }
+
+        [Fact]
+        public void AddApiAuthenticationForMissingApiKeyConfig()
+        {
+            // Arrange
+            IServiceCollection services = new ServiceCollection();
+
+            var inMemorySettings = new Dictionary<string, string> {
+                {"DmrServiceSettings:CentOpsApiKey", "value"},
+                {"DmrServiceSettings:CentOpsUrl", "value"},
+                {"ConnectionStrings:ApiKey", string.Empty},
+            };
+
+            IConfiguration configuration = new ConfigurationBuilder()
+               .AddInMemoryCollection(inMemorySettings)
+               .Build();
+
+            // Act & Assert
+            _ = Assert.Throws<ArgumentException>(() => services.AddApiAuthentication(configuration));
+        }
+
+
+        [Fact]
+        public void AddApiAuthenticationAddsCorrectTypes()
+        {
+            // Arrange
+            IServiceCollection services = new ServiceCollection();
+
+            var inMemorySettings = new Dictionary<string, string> {
+                {"DmrServiceSettings:CentOpsApiKey", "value"},
+                {"DmrServiceSettings:CentOpsUrl", "value"},
+                {"ConnectionStrings:ApiKey", "key"},
+            };
+
+            IConfiguration configuration = new ConfigurationBuilder()
+               .AddInMemoryCollection(inMemorySettings)
+               .Build();
+
+            // Act
+            services.AddApiAuthentication(configuration);
+
+            // Assert
+            // Ensure that there's an ICentOpsService Implementation otherwise code needs to change.
```

Comment:
Did you mean to add this comment? There's no code that seems to be checking for ICentOpsService?